src/elements/select/multiselect.svelte

```diff
@@ -0,0 +1,425 @@
+  optionMatch = false;
+  for (const value of sortedOptions) {
+    if (searchTerm.toLowerCase() === value.toLowerCase()) {
+      optionMatch = true;
+      optionMatchText = value;
+    } 
+  }
+
+};
+
+const handleKeyUp = (event: KeyboardEvent) => {
+  setKeyboardControl(true);
+
+  switch (event.key.toLowerCase()) {
+    case 'enter': return handleEnter();
+    case 'arrowup': return handleNavigate(-1);
+    case 'arrowdown': return handleNavigate(+1);
+    case 'escape': return handleEscape();
+  }
+};
+
+const handleEnter = () => {
+  const option = sortedOptions[navigationIndex]!;
+  value = value.includes(option)
+    ? [...parsedSelected.filter(item => item !== option)].toString()
+    : [...parsedSelected, option].toString();
+  input.focus();
+
+  if (optionMatch) {
+    if (value.includes(optionMatchText)) {
+      value = value.replace(`${optionMatchText},`, '');
+    } else {
+      value += `${optionMatchText},`;
+    }
+    searchTerm = '';
+    optionMatch = false;
+  }
+
+  dispatch('input', { value, values: value.split(',') });
+};
+
+const handleNavigate = (direction: number) => {
+  navigationIndex += direction;
+
+  if (navigationIndex < 0) {
+    navigationIndex = sortedOptions.length - 1;
+  } else if (navigationIndex >= sortedOptions.length) {
+    navigationIndex = 0;
+  }
+
+  const element = optionsContainer.children[0]!.children[navigationIndex]!;
+
+  if (utils.isElementInScrollView(element) === false) {
+    element.scrollIntoView();
+  }
+};
+
+const clearNavigationIndex = () => {
+  navigationIndex = -1;
+};
+
+const handleEscape = () => {
+  input.blur();
+};
+
+const handleFocus = () => {
+  if (open || isDisabled) {
+    return;
+  }
+
+  open = true;
+  input.focus();
+};
+
+const handleFocusOut = (event: FocusEvent) => {
+  if (!root.contains(event.relatedTarget as Node)) {
+    open = false;
+    navigationIndex = -1;
+  }
+};
+
+const handleIconClick = () => {
+  if (!open) {
+    input.focus();
+  } else {
+    open = false;
+  }
+};
+
+const handlePillClick = (target: string) => {
+  value = [...parsedSelected.filter((item: string) => item !== target)].toString();
+  dispatch('input', { value, values: value.split(',') });
+  input.focus();
+};
+
+const handleOptionMouseEnter = (index: number) => {
+  if (keyboardControlling) {
+    return;
+  }
+
+  navigationIndex = index;
+};
+
+const handleOptionSelect = (target: string, event: Event) => {
+  const { checked } = (event.target as HTMLInputElement);
+
+  value = checked
+    ? [...parsedSelected, target].toString()
+    : [...parsedSelected.filter((item: string) => item !== target)].toString();
+
+  input.focus();
+  if (checked) {
+    dispatch('input', { value, values: value.split(','), added: target });
+  } else {
+    dispatch('input', { value, values: value.split(','), removed: target });
+  }
+};
+
+const handleClearAll = () => {
+  value = '';
+  optionsContainer.scrollTop = 0;
+  dispatch('input', { value, values: value.split(',') });
```

Comment:
can we also dispatch another event here in addition to this called `clear-all-click`? since we don't return which values have been removed like in the other `input` events, the regular `input` event doesn't have enough info for me to write a generic `handleInput` function that accounts for clearing all values. if i attach a separate listener for a clear all event, I can handle that specific scenario independently of input changing

Reply:
sure! will add